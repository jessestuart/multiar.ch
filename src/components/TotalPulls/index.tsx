import { graphql, useStaticQuery } from 'gatsby'
import _ from 'lodash'
import fp from 'lodash/fp'
import React from 'react'
import { Box, Text } from 'rebass/styled-components'

import { colors } from 'styles/Theme'

const TotalPulls = () => {
  const query = useStaticQuery(graphql`
    {
      allDockerHubRepo {
        edges {
          node {
            pullCount
          }
        }
      }
    }
  `)
  const totalPulls: number = _.flow(
    fp.get('allDockerHubRepo.edges'),
    fp.sumBy('node.pullCount'),
  )(query)

  return (
    <Box className="flex justify-center items-center">
      Total pulls:
      <Text className="pl1 b" style={{ color: colors.text }}>
        {totalPulls}
      </Text>
    </Box>
  )
}

export default TotalPulls
