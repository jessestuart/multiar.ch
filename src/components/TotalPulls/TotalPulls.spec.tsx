import gatsby from 'gatsby'
import _ from 'lodash'
import React from 'react'
import renderer from 'react-test-renderer'

import TotalPulls from '.'
import fixtures from '../../../test/fixtures/DockerHubReposFixture'

jest.mock('gatsby', () => ({
  graphql: jest.fn(),
  useStaticQuery: jest.fn(),
}))

describe('TotalPulls component.', () => {
  beforeEach(() => {
    // @ts-ignore
    gatsby.useStaticQuery = jest.fn().mockImplementation(() => ({
      allDockerHubRepo: {
        edges: fixtures.map(fixture => ({
          node: fixture,
        })),
      },
    }))
  })

  test('Render TotalPulls component.', () => {
    const component = renderer.create(<TotalPulls />)
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
