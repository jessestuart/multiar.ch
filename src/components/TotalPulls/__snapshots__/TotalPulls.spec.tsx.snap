// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TotalPulls component. Render TotalPulls component. 1`] = `
<div
  className="flex justify-center items-center sc-bdVaJa gtVQmG"
>
  Total pulls:
  <div
    className="pl1 b sc-bdVaJa gtVQmG"
    style={
      Object {
        "color": "rgba(0, 0, 0, 0.75)",
      }
    }
  >
    1724122
  </div>
</div>
`;
